packages/amplify-graphql-model-transformer/src/resources/iterative-dynamo-model-resource-generator.ts

```diff
@@ -0,0 +1,260 @@
+      type: 'String',
+      default: 'true',
+      allowedValues: ['true', 'false'],
+    });
+    // // add the connection between the root and nested stack so the values can be passed down
+    (scope as cdk.NestedStack).setParameter(readIops.node.id, cdk.Fn.ref(ResourceConstants.PARAMETERS.DynamoDBModelTableReadIOPS));
+    (scope as cdk.NestedStack).setParameter(writeIops.node.id, cdk.Fn.ref(ResourceConstants.PARAMETERS.DynamoDBModelTableWriteIOPS));
+    (scope as cdk.NestedStack).setParameter(billingMode.node.id, cdk.Fn.ref(ResourceConstants.PARAMETERS.DynamoDBBillingMode));
+    (scope as cdk.NestedStack).setParameter(
+      pointInTimeRecovery.node.id,
+      cdk.Fn.ref(ResourceConstants.PARAMETERS.DynamoDBEnablePointInTimeRecovery),
+    );
+    (scope as cdk.NestedStack).setParameter(enableSSE.node.id, cdk.Fn.ref(ResourceConstants.PARAMETERS.DynamoDBEnableServerSideEncryption));
+
+    // // Add conditions.
+    new cdk.CfnCondition(scope, ResourceConstants.CONDITIONS.HasEnvironmentParameter, {
+      expression: cdk.Fn.conditionNot(cdk.Fn.conditionEquals(context.synthParameters.amplifyEnvironmentName, ResourceConstants.NONE)),
+    });
+    const useSSE = new cdk.CfnCondition(scope, ResourceConstants.CONDITIONS.ShouldUseServerSideEncryption, {
+      expression: cdk.Fn.conditionEquals(enableSSE, 'true'),
+    });
+    const usePayPerRequestBilling = new cdk.CfnCondition(scope, ResourceConstants.CONDITIONS.ShouldUsePayPerRequestBilling, {
+      expression: cdk.Fn.conditionEquals(billingMode, 'PAY_PER_REQUEST'),
+    });
+    const usePointInTimeRecovery = new cdk.CfnCondition(scope, ResourceConstants.CONDITIONS.ShouldUsePointInTimeRecovery, {
+      expression: cdk.Fn.conditionEquals(pointInTimeRecovery, 'true'),
+    });
+
+    const removalPolicy = this.options.EnableDeletionProtection ? cdk.RemovalPolicy.RETAIN : cdk.RemovalPolicy.DESTROY;
+
+    // TODO: The following lines for custom resource creation will be wrapped into a new internal construct
+    // TODO: The attribute of encryption and TTL should be added
+    const defaultTableState = {
+      TableName: tableName,
+      AttributeDefinitions: [
+        {
+          AttributeName: 'id',
+          AttributeType: 'S',
+        },
+      ],
+      KeySchema: [
+        {
+          AttributeName: 'id',
+          KeyType: 'HASH',
+        },
+      ],
+      StreamSpecification: {
+        StreamViewType: 'NEW_AND_OLD_IMAGES',
+      },
+    };
+    const tableResource = new CustomResource(scope, tableLogicalName, {
+      serviceToken: this.customResourceServiceToken,
+      resourceType: CUSTOM_DDB_CFN_TYPE,
+      properties: {
+        ...defaultTableState,
+      },
+      removalPolicy,
+    });
+    setResourceName(tableResource, { name: modelName, setOnDefaultChild: true });
+
+    // construct a wrapper around the custom table to allow normal CDK operations on top of it
+    const table = Table.fromTableAttributes(scope, `CustomTable${tableLogicalName}`, {
```

Comment:
We should pick something a bit more descriptive than "CustomTable" -- you refer to it as "AmplifyManagedTable" in the feature flag. That might get a bit long, but maybe "ManagedTable"?